### PR TITLE
Add backend support for Apple auth provider

### PR DESCRIPTION
**Pull request by paskal**
It's a bit different from other OAuth providers and requires a different set of options and a private key file.

Backend part of #906.

### CODE REVIEW

File: site/src/docs/configuration/authorization/index.md [hunk 1]
```diff
@@ -6,6 +6,23 @@ title: Authorization
...
+1. Configure "Sign in with Apple" for created Service ID. Add domain where you will use that auth on to "Domains and subdomains" and its main page URL (like `https://example.com/` to "Return URLs".
+1. Register a [New Key](https://developer.apple.com/account/resources/authkeys/list) (private key) for the "Sign in with Apple" feature and download it. Write down the **Key ID**. This key will be used to create [JWT](https://developer.apple.com/documentation/sign_in_with_apple/generate_and_validate_tokens#3262048) Client Secret.
+1. Add your Remark42 domain name and sender email in the Certificates, Identifiers & Profiles >> [More](https://developer.apple.com/account/resources/services/configure) section as a new Email Source.
+
+After completing the previous steps, you can proceed with configuring the Apple auth provider. You'll need to set the following environment variables:
+
+- `AUTH_APPLE_CID` (**required**) - Apple client ID
+- `AUTH_APPLE_TID` (**required**) - Apple service ID
+- `AUTH_APPLE_KID` (**required**) - Private key ID
+- `AUTH_APPLE_PRIVATE_KEY_FILEPATH` (default `/var/apple.p8`) - Private key file location
```
Comment (umputun) [inline site/src/docs/configuration/authorization/index.md:24]: unless I'm missing it, I don't see much info about where  this private key is coming from

File: site/src/docs/configuration/authorization/index.md [hunk 2]
```diff
@@ -6,6 +6,23 @@ title: Authorization
 
 Authentication handled by external providers. You should set up OAuth2 for at least one of them to allow users to make comments. It is not mandatory to have all of them, but one should be correctly configured.
 
+### Apple
```
Comment (umputun) [inline site/src/docs/configuration/authorization/index.md:9]: maybe we should add "not fully implemented yet", as this thing is not yet working on the frontend side?

File: site/src/docs/configuration/authorization/index.md [hunk 3]
```diff
@@ -6,6 +6,23 @@ title: Authorization
...
+1. Configure "Sign in with Apple" for created Service ID. Add domain where you will use that auth on to "Domains and subdomains" and its main page URL (like `https://example.com/` to "Return URLs".
+1. Register a [New Key](https://developer.apple.com/account/resources/authkeys/list) (private key) for the "Sign in with Apple" feature and download it. Write down the **Key ID**. This key will be used to create [JWT](https://developer.apple.com/documentation/sign_in_with_apple/generate_and_validate_tokens#3262048) Client Secret.
+1. Add your Remark42 domain name and sender email in the Certificates, Identifiers & Profiles >> [More](https://developer.apple.com/account/resources/services/configure) section as a new Email Source.
+
+After completing the previous steps, you can proceed with configuring the Apple auth provider. You'll need to set the following environment variables:
+
+- `AUTH_APPLE_CID` (**required**) - Apple client ID
+- `AUTH_APPLE_TID` (**required**) - Apple service ID
```
Comment (umputun) [inline site/src/docs/configuration/authorization/index.md:22]: just wonder, why TID and not SID ?

**Comment by paskal** [comment]
I've altered the instructions to clarify them, here is the current version:
image

**Review by umputun — APPROVED** [review]

**Pull request merged**